test/functional/lua/fs_spec.lua

```diff
@@ -432,6 +432,13 @@ describe('vim.fs', function()
       command('file lua://')
       eq(test_source_path, exec_lua([[return vim.fs.root(0, 'CMakePresets.json')]]))
     end)
+
+    it('returns an absolute path for an invalid filename', function()
+      command('new')
+      command('set buftype=nofile')
+      command('file lua://')
+      eq(test_source_path, exec_lua([[return vim.fs.root('file://asd', 'CMakePresets.json')]]))
```

Comment:
this test is not actually asserting a full path...

Reply:
Thanks for taking a look!

I'm not sure I understand your comment.

Before my change, `vim.fs.root('file:/asd')` would return `.` and after my change it returns the absolute path for the current working directory instead. For the test, `test_source_path` matches the absolute path for the current test and it passes with the change but fails with original code (because `.` doesn't match).

Maybe I'm missing something?

Reply:
yes, I just meant that the test should actually check that `test_source_path` is absolute. see test update